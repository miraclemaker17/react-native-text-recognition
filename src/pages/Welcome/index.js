import React, { Component } from 'react';
import { StackActions, NavigationActions } from 'react-navigation';
import PropTypes from 'prop-types';

import { StatusBar, ActivityIndicator, AsyncStorage } from 'react-native';

import { Container, Title, TextInformation, Error, Form, Input, Button, ButtonText} from './styles'

export default class Welcome extends Component {

    constructor() {
        super();
    
        this.signIn = this.signIn.bind(this);
    }

    static navigationOptions = {

        header: null,
    
    };
    

    static propTypes = {

        navigation: PropTypes.shape({
            dispatch: PropTypes.func,
        }).isRequired,

    };

    state = {

        username: '',
        loading: false,
        errorMessage: null,
        
    }

    saveUser = async (username) => {

        await AsyncStorage.setItem('@FirebaseText:username', username)

    }

    async signIn () {

        const { username } = this.state;

        if(username.length === 0 ) return;

        this.setState({ loading: true });

        try {

            await this.saveUser(username);

            const resetAction = StackActions.reset ({

                index: 0,
                actions: [
    
                    NavigationActions.navigate({ routeName: 'Main' }),
    
                ]
    
            });
    
            this.props.navigation.dispatch(resetAction);

        } catch(err) {

            this.setState({ loading: false, errorMessage: 'Usuário não existe' });

        }

    }

    render() {
      return (

        <Container>

            <StatusBar barStyle="light-content" /> 

            <Title>Bem-vindo</Title>
            <TextInformation>
        
                Para continuar, precisamos que você informe seu usuário
        
            </TextInformation>

            { !!this.state.errorMessage 
                && <Error>{this.state.errorMessage}</Error> }

            <Form>

                <Input

                    autoCapitalize="none"
                    autoCorrect={false}
                    placeholder="Digite seu usuário"
                    underlineColorAndroid='rgba(0, 0, 0, 0)'
                    value = {this.state.username}
                    onChangeText={username => this.setState({ username })}

                />

                <Button onPress={this.signIn}>

                    { this.state.loading
                    ? <ActivityIndicator size="small" color='#FFF' />
                    : <ButtonText>Prosseguir</ButtonText> }

                </Button>

            </Form>

        </Container>

      )
    }
};