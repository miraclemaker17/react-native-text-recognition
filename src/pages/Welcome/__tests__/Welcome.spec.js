import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Welcome from '../index';

const props = {

    navigation: {

        navigate: jest.fn(),
        dispatch: jest.fn()

    }

};

it('The default state is correctly', () => {
    const wrapper = shallow(<Welcome {...props} />);

    expect(wrapper.state().username).toBeDefined();
    expect(wrapper.state().loading).toBeDefined();
    expect(wrapper.state().errorMessage).toBeDefined();

  });

  it('saveUser is called', () => {
    const wrapper = shallow(<Welcome {...props} />);

    const spy = jest.spyOn(wrapper.instance(), 'saveUser');

    wrapper.update();
    wrapper.instance().saveUser();

    expect(spy).toHaveBeenCalled();
  });

  it('signIn is called', () => {
    const wrapper = shallow(<Welcome {...props} />);

    const spy = jest.spyOn(wrapper.instance(), 'signIn');

    wrapper.update();
    wrapper.instance().signIn();

    expect(spy).toHaveBeenCalled();
  });

  it('signIn was called when button click', () => {
    const spy = jest.spyOn(Welcome.prototype, 'signIn');

    const wrapper = shallow(<Welcome {...props} />);

    wrapper.setState({ loading: false })
    
    wrapper.update();

    // wrapper.find('TouchableOpacity').simulate('click');

    const nextButton = wrapper.find('TouchableOpacity').first();

    // Enzyme usually allows wrapper.simulate() alternatively, but this doesn't support 'press' events.
    nextButton.props().onPress();

    expect(spy).toHaveBeenCalled();

  });

  it('saveUser is called in signIn', () => {
    const wrapper = shallow(<Welcome {...props} />);

    wrapper.setState({ username: 'asassasdds' })

    const spy = jest.spyOn(wrapper.instance(), 'saveUser');

    wrapper.update();

    wrapper.instance().saveUser()

    expect(spy).toHaveBeenCalled();

  });

it('renders correctly', () => {
    const wrapper = shallow(<Welcome {...props} />);
    
    expect(wrapper).toMatchSnapshot()
});