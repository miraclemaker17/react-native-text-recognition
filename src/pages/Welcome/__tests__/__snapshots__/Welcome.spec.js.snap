// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#444A5A",
      "flex": 1,
      "justifyContent": "center",
      "padding": 40,
    }
  }
>
  <StatusBar
    animated={false}
    barStyle="light-content"
    showHideTransition="fade"
  />
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 24,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Bem-vindo
  </Text>
  <Text
    style={
      Object {
        "color": "#DDD",
        "fontSize": 14,
        "lineHeight": 21,
        "marginTop": 10,
        "textAlign": "center",
      }
    }
  >
    Para continuar, precisamos que você informe seu usuário
  </Text>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Digite seu usuário"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 3,
          "height": 44,
          "paddingHorizontal": 20,
        }
      }
      underlineColorAndroid="rgba(0, 0, 0, 0)"
      value=""
    />
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#7A91CA",
          "borderRadius": 3,
          "height": 44,
          "justifyContent": "center",
          "marginTop": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        Prosseguir
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;
